client/tests/unit/CosmologyForm.spec.js

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable no-await-in-loop, no-restricted-syntax, no-continue */
+
+import { mount, createLocalVue } from '@vue/test-utils';
+import CosmologyForm from '@/components/forms/CosmologyForm';
+import BACKEND_CONSTANTS from '@/constants/backend_constants';
+import INITIAL_STATE from '@/constants/initial_state';
+
+describe('Mounted CosmologyForm', () => {
+  const localVue = createLocalVue();
+  const wrapper = mount(CosmologyForm, {
+    propsData: {
+      title: 'Cosmology',
+      init: INITIAL_STATE.cosmology,
+    },
+    localVue,
+  });
+
+  const options = Object.keys(BACKEND_CONSTANTS.Cosmo_params);
+
+  test('has correct default model', async () => {
+    await wrapper.vm.$nextTick();
+    expect(wrapper.vm.model.cosmo_model).toBe('Planck15');
+  });
+
+  test('changes model parameters when model is changed', async () => {
+    for (const option of options) {
+      if (wrapper.vm.model.cosmo_model === option) continue;
+      const oldParams = wrapper.vm.model.cosmo_params;
+      wrapper.vm.$data.model.cosmo_model = option;
+      await localVue.nextTick();
+      await localVue.nextTick();
+      const newParams = wrapper.vm.model.cosmo_params;
+      expect(oldParams).not.toBe(newParams);
+    }
+  });
+
+  test('renders correct fields for each model selection', async () => {
+    for (const option of options) {
+      wrapper.vm.$data.model.cosmo_model = option;
+      await localVue.nextTick();
+      await localVue.nextTick();
+      const params = Object.keys(BACKEND_CONSTANTS.Cosmo_params[option].cosmo_params);
+      for (const param of params) {
+        switch (param) {
+          case 'H0':
+            expect(wrapper.html()).toEqual(expect.stringMatching(new RegExp('.*H<sub>0</sub>.*')));
+            break;
+          case 'Ob0':
+            expect(wrapper.html()).toEqual(expect.stringMatching(new RegExp('.*Ω<sub>b</sub>.*')));
+            break;
+          case 'Om0':
+            expect(wrapper.html()).toEqual(expect.stringMatching(new RegExp('.*Ω<sub>m</sub>.*')));
+            break;
+          case 'z':
+            expect(wrapper.html()).toEqual(expect.stringMatching(new RegExp('.*Redshift.*')));
+            break;
+          case 'n':
+            expect(wrapper.html()).toEqual(expect.stringMatching(new RegExp('.*n<sub>s</sub>.*')));
+            break;
+          case 'sigma_8':
+            expect(wrapper.html()).toEqual(expect.stringMatching(new RegExp('.*σ<sub>8</sub>.*')));
+            break;
+          default:
+            expect(wrapper.html()).toEqual(expect.stringMatching(new RegExp(`.*${param}.*`)));
+            break;
+        }
```

Comment:
Not DRY. 0 reason for the switch. Make a list of the tags and loop through that instead of params -- the 6 you have cases for are the only ones that appear for this model.

['H<sub>0</sub>', '*Ω<sub>b</sub>', etc...]